src/utils.ts

```diff
@@ -0,0 +1,456 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {
+  Block,
+  ApiResponse,
+  FullBlock,
+  Input,
+  Output,
+  DecodedScript,
+  FullTx,
+  Token,
+  StatusEvent,
+} from './types';
+import AWS from 'aws-sdk';
+import axios from 'axios';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+AWS.config.update({
+  region: 'us-east-1',
+});
+
+// const FULLNODE_URL = process.env.FULLNODE_URL || 'ws://localhost:8080/v1a/ws/';
+const DEFAULT_SERVER = process.env.DEFAULT_SERVER || 'https://node1.foxtrot.testnet.hathor.network/v1a/';
+const TOKEN_INDEX_MASK = 0b01111111;
+
+let globalCache = {};
+
+export const lambdaCall = (fnName: string, payload: any): Promise<any> => new Promise((resolve, reject) => {
+  const lambda = new AWS.Lambda({
+    apiVersion: '2015-03-31',
+    endpoint: process.env.STAGE === 'local'
+      ? 'http://localhost:3002'
+      : `https://lambda.${process.env.AWS_REGION}.amazonaws.com`,
+  });
+
+  const params = {
+    FunctionName: `${process.env.SERVICE_NAME}-${process.env.STAGE}-${fnName}`,
+    Payload: JSON.stringify({
+      body: payload,
+    }),
+  };
+
+  lambda.invoke(params, (err, data) => {
+    if (err) {
+      console.log('err', err);
+      reject(err);
+    } else {
+      if (data.StatusCode !== 200) {
+        reject(new Error('Request failed.'));
+      }
+
+      try {
+        const responsePayload = JSON.parse(data.Payload as string);
+        const body = JSON.parse(responsePayload.body);
+
+        resolve(body);
+      } catch(e) {
+        console.log('Erroed parsing response body: ', data.Payload);
+
+        return reject(e.message);
+      }
+    }
+  });
+});
+
+export const sendTx = async (tx): Promise<ApiResponse> => {
+  const response = await lambdaCall('onNewTxRequest', tx);
+
+  return response;
+};
+
+export const getWalletServiceBestBlock = async (): Promise<Block> => {
+  const response = await lambdaCall('getLatestBlock', {});
+  const bestBlock: Block = response.block;
+
+  return bestBlock;
+};
+
+export const getFullNodeBestBlock = async (): Promise<Block> => {
+  const response = await axios.get(`${DEFAULT_SERVER}transaction?type=block&count=1`);
+  const { transactions } = response.data;
+
+  const bestBlock: Block = {
+    txId: transactions[0].tx_id as string,
+    height: transactions[0].height as number,
+  };
+
+  return bestBlock;
+};
+
+export const downloadTx = async (txId: string, noCache: boolean = false) => {
+  if (!noCache && globalCache[txId]) {
+    return globalCache[txId];
+  }
+
+  const response = await axios.get(`${DEFAULT_SERVER}transaction?id=${txId}`);
+
+  if (Object.keys(globalCache).length > 200) {
```

Comment:
I clean the cache because transactions are big and they are being kept in memory

I updated the cache to use an LRU cache, added tests and set the `TX_CACHE_SIZE` as a env variable

Done in a168187ca0a72a9e4079f7968d5607fe712e1557